### PR TITLE
Added: Redux

### PR DESCRIPTION
**Pull request by hussiantabha**
Added redux to all the main features

### CODE REVIEW

File: src/components/ListingVideos.js [hunk 1]
```diff
@@ -1,42 +1,51 @@
...
-                  <span className="video-duration-badge">{duration}</span>
-                </div>
-                <div className="video-content-container">
-                  <img src={avatar_url} className="creator-avatar" />
-                  <div className="video-text-container">
-                    <h3>{title}</h3>
-                    <p>{creator}</p>
-                    <p>{`${views} views . ${uploaded}`}</p>
+      {isLoading ? (
+        <h1>Loading</h1>
```
Comment (junaidshaikh-js) [inline src/components/ListingVideos.js:16]: A loader can be shown here instead of loading text

File: src/features/like.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+);
+const likeSlice = createSlice({
+  name: "like",
+  initialState,
+  reducers: {
+    addLikeVideo: (state, action) => {
+      state.likeVideos = action.payload.value;
+    },
+    deleteLikeVideoReducer: (state, action) => {
+      state.likeVideos = action.payload.value;
```
Comment (junaidshaikh-js) [inline src/features/like.js]: can these both reducers join into one such as `updateLikeVideo` as they are doing the same thing?

File: src/features/playlist.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+const initialState = {
+  playlist: [],
+  isLoading: false,
+};
+// const token =
+//   "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiIxMjVkM2MzNy01MjcwLTQ5NjgtODQ0MC1iZTM3ZDFhNTE5OGUiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.DPS9hLIaykSx9V9SwXsOhWgWQ7nk8MtTyumcWlbYamM";
```
Comment (junaidshaikh-js) [inline src/features/playlist.js]: remove code comments

File: src/features/playlist.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  initialState,
+  reducers: {
+    addPlaylistData: (state, action) => {
+      state.playlist = action.payload.value;
+    },
+    deletePlaylistReducer: (state, action) => {
+      state.playlist = action.payload.value;
+    },
+    deletePlaylistVideoReducer: (state, action) => {
+      console.log(action);
```
Comment (junaidshaikh-js) [inline src/features/playlist.js]: remove console logs

File: src/store.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { configureStore } from "@reduxjs/toolkit";
+import videoReducers from "./features/video";
+import playlistReducers from "./features/playlist";
+import likeReducers from "./features/like";
+import watchLaterReducers from "./features/watchLater";
+import historyReducers from "./features/history";
+import loginReducers from "./features/login";
```
Comment (junaidshaikh-js) [inline src/store.js:7]: an `index.js` file inside the feature can be used to export all the features from a single file.

**Review by junaidshaikh-js — COMMENTED** [review]
Great work! 

Suggestion: slice file names can be changed 

for example, `history.js` => `historySlice.js`

**Review by paarthbajaj — APPROVED** [review]
Small PR. Looks good and structured. You can merge.

**Pull request merged**